### PR TITLE
improve naming and comments on resource shard management

### PR DESCRIPTION
**Pull request by vangheem**
### Description
When reviewing this code, I think this naming would have helped me understand what was going on more quickly.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb/nucliadb/ingest/orm/abc.py [hunk 1]
```diff
@@ -50,23 +50,25 @@
 
 class ShardCounter(BaseModel):
     shard: str
-    fields: int
+    resources: int
```
Comment (lferran) [inline nucliadb/nucliadb/ingest/orm/abc.py:53]: mh I am not sure about this one.
Resources can have many fields mapped to it, so I think it makes sense to shard based on the number of fields

File: nucliadb/nucliadb/ingest/settings.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ class Settings(BaseSettings):
     sidecar_port_map: Dict[int, int] = {}
 
     # Node limits
-    max_node_fields: int = 200000
+    max_shard_fields: int = 200000  # max number of fields to target per shard
     max_node_shards: int = 600
```
Comment (lferran) [inline nucliadb/nucliadb/ingest/settings.py:92]: Maybe that should be named `max_node_replicas` then?

From ingest point of view (and in my mind) a shard is a logic shard. A replica shard is the container unit from the node's point of view

**Review by lferran — APPROVED** [review]

File: nucliadb/nucliadb/ingest/settings.py (hunk 2 above)
Comment (vangheem) [reply nucliadb/nucliadb/ingest/settings.py:92]: it probably shouldn't be "max" either then right? It's the number that we always use. So just "shard_replicas"?

Comment (lferran) [reply nucliadb/nucliadb/ingest/settings.py:92]: Just to clarify: `max_node_shards` here does not mean the number of replicas for a given shard.
It means the maximum number of shard replicas that a node can have

Comment (vangheem) [reply nucliadb/nucliadb/ingest/settings.py:92]: yeah, I'm dumb. I see now. I think it makes sense

**Pull request merged**